src/hotspot/share/compiler/compilerDefinitions.cpp

```diff
@@ -524,6 +523,13 @@ bool CompilerConfig::check_args_consistency(bool status) {
 #endif
   }
 
+  // TieredStopAtLevel==0 allocates nmethod space in the code heap with
```

Comment:
I tested with this setting and it works fine with -XX:+SegmentedCodeCache because the code for segmented code cache only checks Arguments::is_interpreter_only().  This doesn't seem right to me, but the testing also expects this to work.  I could fix the tests to not do this. Now I don't know what anyone wants!